demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,361 @@
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The transport send function that defines the transport interface.
+ *
+ * @param[in] pContext User defined context (TCP socket for this demo).
+ * @param[in] pBuffer Buffer to write to the network stack.
+ * @param[in] bytesToWrite Number of bytes to write to the network.
+ *
+ * @return Number of bytes sent; negative value on error.
+ */
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
+                              const void * pBuffer,
+                              size_t bytesToSend )
+{
+    return ( int32_t ) send( pContext->tcpSocket, pBuffer, bytesToSend, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The transport receive function that defines the transport interface.
+ *
+ * @param[in] tcpSocket User defined context (TCP socket for this demo).
+ * @param[out] pBuffer Buffer to read network data into.
+ * @param[in] bytesToRead Number of bytes requested from the network.
+ *
+ * @return Number of bytes received; negative value on error.
+ */
+static int32_t transportRecv( HTTPNetworkContext_t * pContext,
+                              void * pBuffer,
+                              size_t bytesToRead )
+{
+    return ( int32_t ) recv( pContext->tcpSocket, pBuffer, bytesToRead, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Send an HTTP request based on a specified method and path.
+ *
+ * @param[in] tcpSocket TCP socket.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ *
+ * @return #HTTP_SUCCESS if successful.
+ */
+static HTTPStatus_t _sendHttpRequest( HTTPNetworkContext_t * pContext,
+                                      const char * pMethod,
+                                      const char * pPath )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    HTTPRequestHeaders_t requestHeaders = { 0 };
+    HTTPRequestInfo_t requestInfo = { 0 };
+    HTTPTransportInterface_t transport = { 0 };
+    HTTPResponse_t response = { 0 };
+
+    SdkLog( "Sending HTTP %s request to %s%s\r\n",
+            pMethod, SERVER, pPath );
+
+    /* Initialize the request object. */
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = sizeof( pMethod ) - 1;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = sizeof( pPath ) - 1;
+
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    returnStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                        &requestInfo );
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        /* Define the transport interface. */
+        transport.recv = transportRecv;
+        transport.send = transportSend;
+        transport.pContext = pContext;
+
+        /* Initialize the response object. */
```

Comment:
Can add a comment to mention with a Note that the buffer is being _re-used_ here,